tests/kernel/arch_interface/misc/src/main.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <ztest.h>
+
+#define SHIFT_BITS	10
+#define TEST_ROUND	(1 << SHIFT_BITS)
+
+/**
+ * @brief Test how much time arch_nop() costs
+ *
+ * @details This test try to measure the how many instruction cycle
+ * that arch_nop() will cost in different platform.
+ *
+ * @ingroup kernel_common_tests
+ *
+ * @see arch_nop()
+ */
+static void test_nop(void)
+{
+	uint32_t t_get_time, t_before, t_after, diff;
+
+	t_before = k_cycle_get_32();
+
+	t_after = k_cycle_get_32();
+
+	/* calculate cost time of k_cycle_get_32() */
+	t_get_time = t_after - t_before;
+
+	/* if k_cycle_get_32() cost no time means test not works */
+	if (t_get_time == 0) {
+		ztest_test_skip();
+	}
+
+	t_before = k_cycle_get_32();
+
+	for (int i = 0; i < TEST_ROUND; i++) {
+		arch_nop();
+	}
+
+	t_after = k_cycle_get_32();
+
+	/* calculate cost time of arch_nop() */
```

Comment:
We can say "Calculate delta time of arch_nop()".

Reply:
Got it fix, thanks!